File: LeetCode/935E.java
class Solution {
    public int[] diStringMatch(String S) {
        int largest = S.length();
        int smallest = 0;
        int[] ans = new int[S.length() + 1];
        
        for (int i = 0; i < S.length(); i++){
            if (S.charAt(i) == 'D'){
                ans[i] = largest;
                largest--;
            }
            else {
                ans[i] = smallest;
                smallest++;
            }
        }
        ans[ans.length - 1] = smallest;
        return ans;
    }
}

File: LeetCode/935E.py
class Solution:
    def diStringMatch(self, S):
        """
        :type S: str
        :rtype: List[int]
        """
        lst = []
        largest = len(S)
        smallest = 0
        
        for x in range(len(S)):
            if S[x] == "D":
                lst.append(largest)
                largest -= 1
            else:
                lst.append(smallest)
                smallest += 1
        lst.append(smallest)
        return lst